docs/samples/browser-plugin-meetings/app.js

```diff
@@ -2158,14 +2161,16 @@ function addMedia() {
     preferLiveVideoElm.value = 'Enable';
   }
 
+  currentMediaSettings = getMediaSettings();
```

Comment:
this works, but I think it would be better to update `currentMediaSettings` inside `getMediaSettings()`, because otherwise in the future someone might add another call to `getMediaSettings()` and forget to update `currentMediaSettings`, in fact this is already happening, you have another call to `getMediaSettings()` in joinMeeting() where currentMediaSettings don't get updated and it works fine just by luck, because joinMeeting() then also calls addMedia(). If `currentMediaSettings` are supposed to always reflect the last applied media settings, it would be better to set them inside `getMediaSettings()`

Reply:
actually there is 1 more problem: if user toggles a setting that causes updateMedia() to fail, for example the audio share in a transcoded meeting, then the currentMediaSettings have the wrong value and it causes weird behaviour, because it doesn't take into account the fact that it failed so maybe the best thing to do is to set currentMediaSettings only in the then() clause everywhere after calling addMedia() or updateMedia()

Reply:
@marcin-bazyl Made changes as per suggestion. Also, created a method `setLastMediaSettingToggles()` which will also make the media settings toggles to previous state in case of failure.